### PR TITLE
feat(dapps): show dApps list in wallet connect popup

### LINKED ISSUE #14557: Add WC connected dApps list
### Description
_As a user, I want to know which dApps I am connected to, so that I can manage them in the future (e.g. disconnect)_

The user should see the dApps they're connected to on a specific account. Shouldn't focus on design fidelity, but rather on the correctness of the flow.

[Design link](https://www.figma.com/file/HrmZp1y4S77QJezRFRl6ku/Descoped-dApp-Interactions?type=design&node-id=28-33839&mode=design&t=vDEK7XNRpW1HQBUi-4)

### Acceptance Criteria

- [x] The user should see a list of dApps having active sessions for the current account when pressing the "dApps" button
- [x] The user should see the empty state screen if there are no connections

### PR DESCRIPTION
**Pull request by stefandunca**
### Closes #14557

- [x] Depends on [status-go #51191](https://github.com/status-im/status-go/pull/5191)

Integrate basic functionality for wallet connect in status-go
Update the list of dapps from the SDK
Retrieve the persistence dapps list from the backend as a fallback if there is no connection and SDK can't be initialized
Provide a basic simple view of dapps in the wallet connect popup

https://github.com/status-im/status-desktop/assets/47554641/f0353b3c-e4e8-4daa-abf2-01d8de648156

Other changes:

- `HEAD~1`: feat(dapps) attempt to fix the wallet connect abstraction
- `HEAD~2`: chore(dapps) move responsibility of tracking sessions from WC SDK
- `HEAD~3`: chore(dapp) update SDK interface and the generated code

### CODE REVIEW

**Review by alexjba — COMMENTED** [review]
Nice work! The code looks good to me! 👍 some minor questions inline 
I'll move to testing and will come back with the results

**Review by alexjba — APPROVED** [review]
Tested! I only have the non-blocking questions from the previous review.

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    readonly property alias sessionsModel: d.sessionsModel
+
+    function updatePairings() {
+        d.resetPairingsModel()
+    }
+    function updateSessions() {
+        d.resetSessionsModel()
+    }
+
+    readonly property QtObject _d: QtObject {
```
[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml]: I use it now. Thanks for reminding me.

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    }
+                });
+            })();
+        }
+
+        function getPairingTopicFromPairingUrl(url)
+        {
+            if (!url.startsWith("wc:"))
+            {
+                return null;
```
[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml]: Yes, but not for this utility function instead where we use it.

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            Backpressure.debounce(this, 250, () => {
+                sdk.getActiveSessions((sessionList) => {
+                    for (var topic of Object.keys(sessionList)) {
+                        sessions.append(sessionList[topic]);
+                    }
+                });
+            })();
+        }
+
+        function getPairingTopicFromPairingUrl(url)
```
[the remark this answers is not in the document]
Comment (stefandunca) [reply ui/app/AppLayouts/Wallet/services/dapps/DAppsListProvider.qml]: It was used in the POC and planned to be used for checking if the URL was already used. I added a `TODO #14676`. Planned to be used with the work here https://github.com/status-im/status-desktop/issues/14676

File: ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml [hunk 4]
```diff
@@ -230,23 +234,52 @@ StatusDialog {
...
+        Rectangle {
             Layout.alignment: Qt.AlignHCenter
-            Layout.bottomMargin: 16
+            Layout.preferredWidth: 72
+            Layout.preferredHeight: 72
+
+            radius: width / 2
+            color: Theme.palette.primaryColor3
+
+            StatusImage {
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/ConnectDAppModal.qml:247]: `StatusRoundedImage`? I assume that's why you used the `OpacityMask` below

**Review by caybro — APPROVED** [review]
LGTM

**Review by lukaszso — APPROVED** [review]
@stefandunca tested, i followed the acceptance criteria from the original issue and LGTM.

**Pull request merged**